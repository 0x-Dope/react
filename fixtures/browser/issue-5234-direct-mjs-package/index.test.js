const { bootstrap, startDevelopmentServer } = require('../../utils');
const puppeteer = require('puppeteer');

beforeEach(async () => {
  await bootstrap({ directory: global.testDirectory, template: __dirname });
  global.appPort = await startDevelopmentServer({
    directory: global.testDirectory,
  });
  await new Promise(resolve => setTimeout(resolve, 3000));
});

// https://github.com/facebook/create-react-app/issues/5234
describe('issue #5234 (mjs files are imported as static files)', () => {
  it('correctly bundles files in development', async () => {
    const browser = await puppeteer.launch({ headless: true });
    try {
      const page = await browser.newPage();
      console.log(`http://localhost:${global.appPort}/`);
      await page.goto(`http://localhost:${global.appPort}/`);
      await page.waitForSelector('.App-Ready');
      const output = await page.evaluate(() => {
        return document.testData;
      });
      expect(output).toMatchSnapshot();
    } finally {
      browser.close();
    }
  });
});
