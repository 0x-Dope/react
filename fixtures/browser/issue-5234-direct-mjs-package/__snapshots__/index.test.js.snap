// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`issue #5234 (mjs files are imported as static files) correctly bundles files in development 1`] = `
Object {
  "graphql": "undefined",
  "parse": "function",
  "test": "/static/media/Test.4478d87c.mjs",
}
`;
